libs/libarchfpga/src/physical_types_util.h

```diff
@@ -301,4 +302,93 @@ const t_physical_tile_port* get_port_by_pin(const t_sub_tile* sub_tile, int pin)
  */
 const t_port* get_port_by_pin(t_logical_block_type_ptr type, int pin);
 
+
+/************************************ Access to intra-block resources ************************************/
+
+/* Access information related to pin classes */
+
+
+/** get information given class physical num **/
+std::tuple<const t_sub_tile*, int> get_sub_tile_from_class_physical_num(t_physical_tile_type_ptr physical_tile, int physical_class_num);
```

Comment:
If they are related, is it better to define a struct or class?

Reply:
I agree that if the number of variables exceeds 2, it could be hard to work with it. I changed the functions which return the tuple object with more than 2 variables. For this particular function, I don't think we really need to change it. However, if you still think it should be changed please let me know.